src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -49,57 +64,52 @@ var canUseMap = (
 );
 
 if (canUseMap) {
-  itemMap = new Map();
-} else {
-  itemByKey = {};
-}
-
-var unmountedIDs = [];
-var rootIDs = [];
-
-// Use non-numeric keys to prevent V8 performance issues:
-// https://github.com/facebook/react/pull/7232
-function getKeyFromID(id) {
-  return '.' + id;
-}
-function getIDFromKey(key) {
-  return parseInt(key.substr(1), 10);
-}
+  var itemMap = new Map();
 
-function get(id) {
-  if (canUseMap) {
+  var get = function(id) {
     return itemMap.get(id);
-  }
-  var key = getKeyFromID(id);
-  return itemByKey[key];
-}
-
-function remove(id) {
-  if (canUseMap) {
+  };
+  var remove = function(id) {
     itemMap.delete(id);
-    return;
-  }
-  var key = getKeyFromID(id);
-  delete itemByKey[key];
-}
-
-function create(id, element, parentID) {
-  var item = {
-    element,
-    parentID,
-    text: null,
-    childIDs: [],
-    isMounted: false,
-    updateCount: 0,
   };
-  if (canUseMap) {
+  var set = function(id, item) {
     itemMap.set(id, item);
-    return;
+  };
+  var keys = function() {
+    return Array.from(itemMap.keys());
+  };
+} else {
+  var itemByKey = {};
+
+  // Use non-numeric keys to prevent V8 performance issues:
+  // https://github.com/facebook/react/pull/7232
+  function getKeyFromID(id: DebugID): string {
```

Comment:
By splitting it, it is now obvious that those two functions are only needed in the object case